spec/controllers/spree/amazon_controller_spec.rb

```diff
@@ -161,7 +161,15 @@ def stub_amazon_order(address: build_amazon_address, email: 'jordan.brough@examp
       stub_amazon_order
       set_current_order(order)
 
-      expect_any_instance_of(SpreeAmazon::Order).to receive(:save_total)
+      expect_any_instance_of(SpreeAmazon::Order).to(
```

Comment:
You can't use `expect_any_instance_of` with `have_received`.  We could use a spy but that seems more verbose w/o much benefit.

This reads OK to my eyes.  I'd vote that we not be too strict about setup/exec/validation since it's not a strong convention in Solidus itself.